### ISSUE TITLE
Incorrect metric Master.EmbeddedJournalSnapshotLastIndex and Master.JournalEntriesSinceCheckPoint 

### ISSUE THREAD

**Issue by maobaolong** [Alluxio/alluxio#13932]
**Alluxio Version:**
Master branch commit id  601272157bf

**Describe the bug**

The metric Master.EmbeddedJournalSnapshotLastIndex and Master.JournalEntriesSinceCheckPoint are not correct.

![image](https://user-images.githubusercontent.com/17329931/129327373-1f4cf6af-863a-4612-a163-d9d87a31554f.png)

![image](https://user-images.githubusercontent.com/17329931/129327785-cff2cfd2-acb7-466f-93b1-7c5ae4e49773.png)

![image](https://user-images.githubusercontent.com/17329931/129328215-3af98618-678d-4b1e-9f95-c29f154e0164.png)

![image](https://user-images.githubusercontent.com/17329931/129328548-0a6a3e1f-2c28-4d81-a8cf-5a394304ef23.png)

![image](https://user-images.githubusercontent.com/17329931/129328705-6d161c72-64fc-4233-86d7-133ac3b2fdce.png)

Log from alluxio primary master, so i guess the current sequence_number is about 5448746

```
2021-08-10 03:03:21,757 WARN  SnapshotReplicationManager - Failed to request snapshot info from followers: alluxio.exception.status.UnavailableException: No recent snapshot found from followers

2021-08-13 14:10:20,105 TRACE RaftJournalWriter - Writing entry 5448746: update_inode {
  id: 19
  last_modification_time_ms: 1628835020097
  last_access_time_ms: 1628835020097
}

2021-08-13 14:10:20,105 TRACE RaftJournalWriter - Flushing entry journal_entries {
  sequence_number: 5448746
  update_inode {
    id: 19
    last_modification_time_ms: 1628835020097
    last_access_time_ms: 1628835020097
  }
} (Message:ba021908aac8cc029a02...(size=28))
2021-08-13 14:10:20,107 TRACE RaftJournalWriter - Writing entry 5448747: inode_directory_id_generator {
  container_id: 36004
  sequence_number: 152943
}

2021-08-13 14:10:20,108 TRACE RaftJournalWriter - Flushing entry journal_entries {
  sequence_number: 5448747
  inode_directory_id_generator {
    container_id: 36004
    sequence_number: 152943
  }
}
 (Message:ba020f08abc8cc025208...(size=18))
```

**To Reproduce**

Not sure

**Expected behavior**

Correct metric, no warn at this time

**Urgency**

High

**Additional context**

No

**Comment by yuzhu** [comment]
@LuQQiu seems like `mSnapshotLastIndex` is not getting updated even after a snapshot is taken? any ideas?

**Comment by maobaolong** [comment]
![image](https://user-images.githubusercontent.com/17329931/129839299-675d2170-c043-4bea-bea9-be2364f209de.png)

As we see the following logs, I suggest to use `downloaded.getIndex()` as lastSnapshotIndex.
```
2021-08-10 03:03:23,598 INFO  SnapshotReplicationManager - Completed storing snapshot at (t:4, i:3999999) to file /var/alluxio-share-dir/journal/raft/02511d47-d67c-49a3-9011-abb3109a44c1/sm/snapshot.4_3999999
```

**Comment by LuQQiu** [comment]
@maobaolong Yeah, it make sense to compare downloaded vs current last snapshot index and use whichever larger

**Comment by maobaolong** [comment]
@LuQQiu The metrics `lastAppliedTermIndex.getIndex()` is not correct too.

**Comment by LuQQiu** [comment]
@maobaolong can you give more information about what's the incorrect behavior of `lastAppliedTermIndex.getIndex()` after our fixes?
lastAppliedTermIndex.getIndex() is provided by Ratis and is not controlled by Alluxio.
Did you see a negative value? 
I looked at the code and the logics seem to be correct
